WeakAuras/Conditions.lua

```diff
@@ -162,7 +162,7 @@ function Private.ExecEnv.ScheduleConditionCheck(time, uid, cloneId)
         checkConditions[uid](region);
         Private.ActivateAuraEnvironment()
       end
-    end, time - GetTime())
+    end, (time - GetTime()) / (modRate or 1.0))
```

Comment:
I'm not sure that this is correct. It's been a while since I last looked at modRate, but I think the expirationTime in the state is always correct and modRate basically only affects how remaining time is formatted.

Reply:
Line 278 the test on remainingTime correctly consider modRate
Issue i found is that the condition function is run too early when this test is not true yet

Reply:
Right but this is then the wrong place to fix it. The calculation of nextTime needs to be adjusted. Also in line 278 the usage of modRate is guarded by useModRate, so that should be done too.

Reply:
Changing nextTime was the first thing i tried with the following change, but it's wrong, didn't work correctly

```Lua
    if (cType == "timer" and value) then
      if useModRate then
        recheckCode = "  nextTime = state[" .. trigger .. "] and not state[" .. trigger .. "].paused"
        .. " and state[" .. trigger .. "]" .. string.format("[%q]",  variable)
        .. " and (state[" .. trigger .. "]" .. string.format("[%q]",  variable) .. " - " .. value .. ") / (state[" .. trigger .. "].modRate or 1.0)\n"
      else
        recheckCode = "  nextTime = state[" .. trigger .. "] and not state[" .. trigger .. "].paused"
        .. " and state[" .. trigger .. "]" .. string.format("[%q]",  variable)
        .. " and (state[" .. trigger .. "]" .. string.format("[%q]",  variable) .. " - " .. value .. ")\n"
      end
      recheckCode = recheckCode .. "  if (nextTime and (not recheckTime or nextTime < recheckTime) and nextTime >= now) then\n"
      recheckCode = recheckCode .. "    recheckTime = nextTime\n";
      recheckCode = recheckCode .. "  end\n"
```

Reply:
Becasue:
**(state[" .. trigger .. "]" .. string.format("[%q]",  variable)** is the expirationTime and **value**  the time threshold time.

You need to modify the threshold time by multiplying it with modRate.

Consider this example:
We are at t = 100.
The expirationTime is 150.
The remaining time check is 10.
The modRate is 2.0 (that is time ticks twice as fast)

Then the check must run at 
expirationTime - (modRate * threshold)
=> 150 - 10 * 2
=> 130

Reply:
You are right, pushed updated commit